### ISSUE TITLE
Reused ".irodsEnv.xxxx" files may result in errors and inconsistencies

### ISSUE THREAD

**Issue by AndreyNevolin** [irods/irods#2406]
Hi,

Hung ".irodsEnv.xxxx" (where "xxxx" is a process ID) files may be collected in ".irods" directory. For example, lots of such files exist in my testing environment.

These files make a big issue.

If ".irodsEnv.xxxx" exists and iRODS client machine is restarted, iRODS client process may get PID "xxxx" again. If that happens, incorrect configuration will be loaded from ".irodsEnv.xxxx" and used by the Agent. After that, non-trivial and crazy things may happen.

For example, wrong 'Cwd' may be specified in the file. Even non-existing 'Cwd' may appear there (see bug #2405 for description of iRODS behavior in this case).

Or even worse:
- assume current 'Cwd' is "/coll1/coll2"
- 'Cwd' in the file is "/COLL1/COLL2"
- all collections exist
- file 'my_file' exists in 'coll2' and a file with the same name exists in 'COLL2'
- user does 'iput -f my_file'
- the user intends to overwrite "/coll1/coll2/my_file", but the file "/COLL1/COLL2/my_file" will be overwritten

To summarize, ".irodsEnv.xxxx" files could be very dangerous.

**Comment by trel** [comment]
These files are the only state being stored on the client side.

A best practice in this case is to use full logical paths in all automated scripts (`icd` could be surprising).
